exchange/bitswap/bitswap.go

```diff
@@ -94,6 +86,15 @@ func New(parent context.Context, p peer.ID, network bsnet.BitSwapNetwork,
 
 	// Start up bitswaps async worker routines
 	bs.startWorkers(px, ctx)
+
+	// bind the context and process.
+	// do it over here to avoid closing before all setup is done.
+	go func() {
```

Comment:
Should https://github.com/ipfs/go-ipfs/blob/master/p2p/net/swarm/swarm_dial.go#L420 has this as well?